File: designpattern/reactive_x/reactive_x/src/reactive_x/custom_observable.cpp
// 独自のヘッダ
#include "custom_observable.h" // class_custom_observable

// メンバ関数subscribe
void class_custom_observable::subscribe(interface_observer *observer){

  // オブザーバーの追加.
  observers_.push_back(observer); // observers_.push_backでobserverを追加.

}

// メンバ関数notify
void class_custom_observable::notify(std::string value){

  // オブザーバーを順次実行.
  for (auto itor = observers_.begin(); itor != observers_.end(); itor++){
    (*itor)->next(value); // itor->nextを順次実行.
  }

}

File: rx/IObserver_T/IObserver_T/src/IObserver_T/IObserver_T/CustomObservable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace IObserver_T
{
    class CustomObservable : IObservable<string>
    {
        // オブザーバーリスト
        private List<IObserver<string>> observers = new List<IObserver<string>>();  // IObserver<string>のList observersを生成.

        public IDisposable Subscribe(IObserver<string> observer)
        {
            // オブザーバーの追加.
            observers.Add(observer);    // observers.Addでobserverを追加.
            return null;    // ひとまずnullを返しておく.
            //throw new NotImplementedException();
        }

        // 独自の実行メソッドExecute
        public void Execute(string value)
        {
            // オブザーバーのOnNextを実行.
            foreach (var o in observers)    // observers全てに対して.
            {
                o.OnNext(value);    // o.OnNext(value)を実行.
            }
        }
    }
}
